File: apis/src/main/java/com/beat/apis/booking/application/dto/BookingRefundResponse.java
package com.beat.apis.booking.application.dto;

import com.beat.apis.booking.application.dto.BookingStatusType;
import com.beat.apis.performance.application.dto.BankNameType;

public record BookingRefundResponse(
	long bookingId,
	BookingStatusType bookingStatus,
	BankNameType bankName,
	String accountNumber,
	String accountHolder
) {
	public static BookingRefundResponse of(
		long bookingId,
		BookingStatusType bookingStatus,
		BankNameType bankName,
		String accountNumber,
		String accountHolder) {
		return new BookingRefundResponse(
			bookingId,
			bookingStatus,
			bankName,
			accountNumber,
			accountHolder);
	}
}

File: apis/src/main/kotlin/com/beat/apis/booking/api/response/BookingRefundResponse.kt
package com.beat.apis.booking.api.response

import com.beat.apis.booking.api.type.BookingStatusType
import com.beat.application.frontoffice.booking.result.BookingRefundResult
import com.beat.apis.performance.api.type.BankNameType

@ConsistentCopyVisibility
data class BookingRefundResponse private constructor(
    val bookingId: Long,
    val bookingStatus: BookingStatusType?,
    val bankName: BankNameType?,
    val accountNumber: String?,
    val accountHolder: String?,
) {
    companion object {
        fun from(result: BookingRefundResult): BookingRefundResponse = BookingRefundResponse(
            bookingId = result.bookingId,
            bookingStatus = result.bookingStatus?.let(BookingStatusType::valueOf),
            bankName = result.bankName?.let(BankNameType::valueOf),
            accountNumber = result.accountNumber,
            accountHolder = result.accountHolder,
        )
    }
}
